mkdocs/contrib/search/search_index.py

```diff
@@ -105,7 +108,7 @@ def generate_search_index(self):
         }
         data = json.dumps(page_dicts, sort_keys=True, separators=(',', ':'))
 
-        if self.config['prebuild_index']:
+        if self.config['prebuild_index'] == 'node':
```

Comment:
This should account for users who previously set the config value to `True`. There are two ways to do this:

1. Leave the user provided value in the config and account for it here with:

        if self.config['prebuild_index'] in ['node', True]:

2. Create a custom config validation class which transforms `True` and `False` values to `'node'` and `''` respectively.

Option 1 is obviously less work, but option 2 provides a way to provide clear error messages, including any deprecation warnings to encourage users to update their config files for a graceful deprecation.

Naturally, either way would need some additional tests to ensure the proper results. For option 1 that would mean replicating the existing tests with a set which pass in  `True` or `False` as config values. For option 2, we would simply need some tests for the custom config validation class to ensure it always resulted in a value containing the proper string.

Reply:
I went with a [`Choice` config validation type](https://github.com/mkdocs/mkdocs/pull/1607/commits/d1ed542a396c7b6962c8cdd57d6ebaf588d8b2de) that accepts one of a set of predefined choices, seemed more flexible that a custom made one just for this case.